californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -210,13 +209,19 @@ public void receiveRequest(final Request request, final EndpointReceiver receive
 		boolean duplicate = previous != null;
 
 		if (duplicate) {
-			// assuming addresses changing, request could not only be deduplicated by their address and mid.
-			EndpointContext sourceContext = request.getSourceContext();
-			Request previousRequest = previous.getRequest();
-			EndpointContext previousSourceContext = previousRequest.getSourceContext();
-			// the previous response would be send with its previous context using 
-			// the current request context as connection context
-			duplicate = endpointContextMatcher.isToBeSent(previousSourceContext, sourceContext);
+			// assuming addresses changing, request could not be
+			// deduplicated only by their address and MID.
```

Comment:
should this comment be moved in else block ?

Reply:
Why? It says, that the request (message) maybe from a different peer than the previous message.